besu/src/main/java/org/hyperledger/besu/Runner.java

```diff
@@ -113,18 +114,20 @@ public void awaitStop() {
   @Override
   public void close() throws Exception {
     try {
+      jsonRpc.ifPresent(service -> waitForServiceToStop("jsonRpc", service.stop()));
+      graphQLHttp.ifPresent(service -> waitForServiceToStop("graphQLHttp", service.stop()));
+      websocketRpc.ifPresent(service -> waitForServiceToStop("websocketRpc", service.stop()));
+      metrics.ifPresent(service -> waitForServiceToStop("metrics", service.stop()));
+
+      besuController.getMiningCoordinator().stop();
+      besuController.getMiningCoordinator().awaitStop();
       if (networkRunner.getNetwork().isP2pEnabled()) {
```

Comment:
In my mind, if we start up services in the order A,B,C, then we want to shut them down in the reverse order C,B,A - assuming that there are dependencies from C->B->A.

For the API services, this seems especially important because some of the behavior of those API's may be undefined when interacting with a component that is shutting down.  So, I think we should be shutting down JSON-RPC, etc before we start shutting down the underlying services they depend on.

Reply:
I'm not sure about the `metrics` service.  I started to move that to be the first thing we bring up and the last service we bring down, but then I stopped because I wanted to minimize the somewhat off-topic changes.

Reply:
Didn't have visibility into the startup in this PR.  That makes sense.